src/CoreProvider.ts

```diff
@@ -0,0 +1,21 @@
+import { ChatCore } from "./ChatCore";
+import { ChatConfig, InternalConfig } from "./models";
+
+/**
+ * Provider for the ChatCore library
+ * @public
+ */
+export function ProvideChatCore(config: ChatConfig): ChatCore {
+  return new ChatCore(config);
+}
+
+/**
+ * Provider for the ChatCore library with additional internal-only configuration
+ * @internal
+ */
+export function ProvideChatCoreInternal(
```

Comment:
nit: functions in JS usually don't lead with uppercase letter. can we update this and above to be `provideChatCore`?

Also, we can overload the function by naming them the same and only define one signature

Reply:
Js can't be overloaded and I think we want them separate for @internal